android/source/src/main/kotlin/com/gu/source/presets/typography/Typography.kt

```diff
@@ -6,788 +6,947 @@ import androidx.compose.ui.text.PlatformTextStyle
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.font.FontStyle
 import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.LineHeightStyle
+import androidx.compose.ui.unit.em
 import androidx.compose.ui.unit.sp
-import androidx.compose.ui.unit.times
 import com.gu.source.R
 import com.gu.source.Source
 import com.gu.source.utils.fontFamilyResource
 
+private val BaseTextStyle = TextStyle(
```

Comment:
> Part of me wonders if you can do something similar for each of the styles, so that only the size is overridden, eg define a `HeadlineBold`, a `HeadlineLight` etc

That sounds good. I'll do it in a follow up PR.

> I might even be tempted to shy away from having explicit tokens for each of the size variations and just have the composable use the style and apply the size it wants; although I suspect designers etc might well like having the big screenshots of the different previews and things (which are quite nice).

No, the size is core to the token. The design system defines specific sizes ([doc](https://theguardian.design/2a1e5182b/p/01555f-typography-presets)) with the intention that only they are used in the app. Size can still be overridden in the composable, but the library provides tokens for those explicit sizes.